pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/DistinctCountULLAggregationFunction.java

```diff
@@ -339,8 +339,9 @@ public UltraLogLog extractGroupByResult(GroupByResultHolder groupByResultHolder,
 
   @Override
   public UltraLogLog merge(UltraLogLog intermediateResult1, UltraLogLog intermediateResult2) {
-    int largerP = Math.max(intermediateResult1.getP(), intermediateResult2.getP());
-    UltraLogLog merged = UltraLogLog.create(largerP);
+    // UltraLogLog object doesn't support merge a smaller P object into a larger p object.
+    int smallerP = Math.max(intermediateResult1.getP(), intermediateResult2.getP());
+    UltraLogLog merged = UltraLogLog.create(smallerP);
```

Comment:
Not introduced in this PR, but do we need to create a new one then merge both ULLs in? Could we merge one into the other one?